### PR TITLE
Experimental: Touch up modern CMake with Trilinos

### PR DESCRIPTION
**Pull request by jjwilke**

### CODE REVIEW

**Comment by ndellingwood** [comment]
@jjwilke does this PR replace #2637, or should it follow?

**Comment by jjwilke** [comment]
I am specing out two different proposals for people to make decisions on. Neither of the PRs should be considered the "correct" one yet.

File: cmake/kokkos_tribits.cmake [hunk 1]
```diff
@@ -204,10 +204,15 @@ ENDMACRO()
 
 FUNCTION(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
```
Comment (masterleinad) [inline cmake/kokkos_tribits.cmake]: Are the changes in this file used anywhere? I don't find any call to it that changed.

**Comment by masterleinad** [comment]
@ndellingwood Could you also test this branch?

**Comment by ndellingwood** [comment]
@masterleinad I think @jjwilke is doing some testing of this vs #2637 (@jjwilke please correct me if wrong)

I'd rather hold off spending time on testing until a decision is made on the path forward toward dealing with the issue of forwarding flags to downstream apps using Trilinos.
Do we follow the route this branch intends (modern cmake fixes), where it's my understanding this will break downstream apps which will be disruptive and likely require coordination to help resolve the issues (it's unclear the level of disruption) vs restoring old behavior in Trilinos along the lines of #2637, which allows more time to coordinate with downstream apps and updating Trilinos' build system appropriately?

**Comment by jjwilke** [comment]
Closing for now, but keeping branch open. We need to get release done, which means pursuing PR #2660 as a fix for Trilinos for now.

**Pull request closed without merging**